Wes review 2024-07-15 (#1)
intarchboard/workshop-nemops: IAB-Workshop.md

Those requirements were instrumental in developing first the NETCONF protocol (in the NETCONF Working Group) [RFC6241], the associated YANG data modeling language (in the NETMOD Working Group) [RFC7950], RESTCONF [RFC8040], and most recently CORECONF [I-D.ietf-core-comi].

It has been more than 20 years since that IAB workshop and it is time
to evaluate what has been achieved since the original workshop and
what recommendations and requirements have changed or still need pursing. This new workshop aims to discuss the following key topics:

- Review the outcomes and results of the 2002 workshop (current deployments, state of the art) and identify any operational barriers that prevent these technologies from being widely implemented (limitations, hurdles).

> - Sketch new requirements for future network management operations in a collaborative manner with the industry, network operators and protocol engineers and develop a plan of action and recommendations for the IETF.

Reviewer: ```suggestion
- Sketch new requirements for future network management operations in a collaborative manner with the industry, network operators, and protocol engineers.

 - Develop a plan of action and recommendations for the IETF.
```

Reviewer: Split the sentence for better readability

## Review of the 2002 Workshop Outcomes

> The first focus of the IAB NEMOPS workshop will be to assess the recommendations and requirements documented in RFC3535 in terms of specification completeness, deployment and utilization. Also, this workshop will assess to what extent the RFC3535 recommendations were driving network management efforts within the IETF and influencing other Standards Development Organizations' (SDOs) activities. 

Reviewer: ```suggestion
The first focus of the IAB NEMOPS workshop will be to assess the requirements documented in RFC3535 in terms of specification completeness, deployment, and utilization. Also, this workshop will assess to what extent the RFC3535 recommendations were driving network management efforts within the IETF and influencing other Standards Development Organizations' (SDOs) activities. 
```

Reviewer: 3535 reco are IETF-centric and are mentioned in the sentence right after.

Author: Those are two different notions in the sentences though.  So, IMHO, they should both say "recommendations and requirements" as the first sentence talks about compleness, deployment and utilization of both, and the second is about whether or not both have affected other SDOs or not.

Reviewer: I'm having troubles to see what "deployment, and utilization" is about for a set of reco that were about creating WGs, dedicate resources, and some other IETF matters.

Reviewer 2: Have this issue resolved? My understanding on utilization is about whether specific part of recommendation or requirements have been quoted or referenced by some standard work in other SDOs.

- What alternative (i.e., non-IETF) solutions are being deployed that better fit operators' needs?

> - What recommendations and requirements from RFC3535 have received either or both little implementation or deployment?

Reviewer: ```suggestion
- What requirements from RFC3535 have received either or both little implementation or deployment?
```

Author: disagree with this -- 3535 has both recommendations and requirements and I was trying to make it very clear that both will be discussed/analyzed/etc.  It's overly verbose, but such is the way of the IETF (IMHO).

Reviewer: Fully agree that both need to be discussed/analyzed/etc. 

My concern is related to mixing these two and talk about "deployment, etc." when referring to considerations related to IETF resources (3535 reco). Please note that we do already have the following in the preamble: 

"Assess to what extent the RFC3535 recommendations were driving network management efforts within the IETF"

Author: I think we want to do both though -- we want to see what was done within the IETF, and whether the results of anything done inside the ietf were used outside.  IE, is netconf getting used?  is SNMP getting sunsetted?  Is everyone just switching to REST?  What's the state of the world, and is it diverging from IETF specs and what can the IETF do (now) to improve things.  It involves both the past, the effect internal and the effect externally, right?

## Laying a path for the future of Network Management

Network topologies have become significantly more complex since the 2002 IAB workshop, as have both router and host technologies. Assuming this trend will continue, this workshop's ambition is to lay a new directional foundation for the continued future of network management protocols by gathering new input on what new issues network operators and network management implementers are facing.

Discussion topics will include, but are not limited to:

- Tooling, opensource, experimentation, proof of concept, multi-vendor interoperability test (e.g., EANTC), and system integration

- Data consistency to support richer observability (Data & Knowledge)

- Integration issues with the business layer

- Automation, orchestration, and autonomy 

> Recommendations that may come out of this effort should be helpful to the standardization development organizations including the IETF and the IRTF, as well as the global industry's implementers, operational groups, and other entities.

Reviewer: ```suggestion
Recommendations that may come out of this effort should be helpful to the global industry including the IETF, the IRTF, implementers, operational groups, and other entities.
```

Author: So we don't intend to create advice that might be helpful to SDOs?

Reviewer: other "SDOs" fall under "other entities".  

Actually, the initial wording (and thus what this change tries to revert back) is to put the emphasis on the global industry approach not separating SDOs vs rest of the ecosystem.

Reviewer 2: see SDOs as part of global industry is new to me but fine, traditionally SDOs focus on developing standard work while industry consortium is responsible for get it deployed or implemented.

Reviewer 3: So we don't intend to create advice that might be helpful to SDOs? => yes we do, but you will see that, getting in the higher levels of OPS architecture, more and more opensource will be involved ... where the IETF is the facilitator. Therefore, I agree with Med's proposal

## Out of scope topics

A workshop dedicated to management techniques related to encrypted networks has recently been held [RFC9490]. Another workshop dedicated to the environmental impact of Internet applications and systems has also been held [RFC9547]. These topics are, therefore, out of scope for this workshop. Topics that are clearly in the scope of NMRG in the IRTF are also out of scope. 

## Pre-Workshop Information Gathering & Outreach

In 2002, many network operators attended IETF meetings and participated in network management protocol discussions. Direct IETF participation from network operators has since decreased, with many operators focusing on conferences that are more central to their needs (RIPE, NANOG, APRICOT, AutoConn, etc.).

Under this effort, we expect some workshop participants to stimulate an outreach effort to attract opinions and interest by visiting these other venues and running information and requirement gathering sessions. This series of outreach efforts will result in workshop position papers that will provide valuable summarized feedback to the online NEMOPS Workshop. 

> The mailing list nemops-interest@iab.org will be used for outreach events at other conferences, general announcements, and related discussions before (and after) the workshop is held at the end of 2024.

Reviewer: ```suggestion
The mailing list nemops-interest@iab.org will be used for outreach events at other conferences, general announcements, and related discussions before (and after) the workshop.
```
